tensorflow/c/eager/tape.h

```diff
@@ -216,23 +216,17 @@ void GradientTape<Gradient, BackwardFunction>::RecordOperation(
 template <typename Gradient, typename BackwardFunction>
 void GradientTape<Gradient, BackwardFunction>::DeleteTrace(int64 tensor_id) {
   auto it = tensor_usage_.find(tensor_id);
-  if (it == tensor_usage_.end()) {
-    return;
-  }
-  it->second--;
-  if (it->second != 0) {
-    return;
-  }
-  tensor_usage_.erase(it);
   auto tensor_op_it = tensor_tape_.find(tensor_id);
-  if (tensor_op_it == tensor_tape_.end()) {
-    return;
-  }
   const int64 op_id = tensor_op_it->second;
-  if (op_id == -1) {
-    // Do not delete watched tensors.
+
+  it->second--;
```

Comment:
I didn't quite follow the change here. What is the "optimization" - is it lines of code or some performance argument being made here?

Reply:
Thanks for reviewing my changes, 
Just have found multiple times if-statement was being used to exit function that's why decided to reduce the lines of code and making it with single statement to pass `empty return`. Just have followed the same single statement as it is done in `line no 322`. If you think it is not valid. I will revert it back. Just wanted to reduce the lines of code.

Reply:
I'd suggest we leave this as is. I know this is subjective, but I don't think the 4 lines saved improve readability of the logic (and it doesn't introduce questions like is it okay to use `it->second--` when there was no entry in `tensor_usage_` etc.

Reply:
Thanks! Changes are reverted as per your suggestion. Can we go ahead :)